packages/nextjs/src/app-router/client/useAwaitableNavigate.ts

```diff
@@ -1,46 +1,61 @@
 'use client';
 
-import { useRouter } from 'next/navigation';
-import { useEffect, useRef, useTransition } from 'react';
+import { usePathname, useRouter } from 'next/navigation';
+import { useCallback, useEffect } from 'react';
 
-type NavigateFunction = ReturnType<typeof useRouter>['push'];
+declare global {
+  interface Window {
+    __internal_clerk_navPromisesBuffer: Array<() => void> | undefined;
```

Comment:
Can we prefix the window properties with `clerk` or `__clerk`?

Reply:
All window APIs used by us are prefixed with __clerk so I will do that. Good catch!